consensus/currency.go

```diff
@@ -94,6 +95,12 @@ func (c Currency) RoundDown(n uint64) (y Currency) {
 	return
 }
 
+// ContractTax returns the amount of Currency that would be taxed from a
+// contract payout with value c.
+func (c Currency) ContractTax() Currency {
+	return c.MulFloat(SiafundPortion).RoundDown(SiafundCount)
+}
```

Comment:
yeah, that makes more sense. This is an improvement everywhere except for renter/negotiate.go, because it needs to calculate the tax before creating the contract. So that function got slightly more awkward.